shared_model/interfaces/iroha_internal/transaction_sequence.cpp

```diff
@@ -46,47 +85,33 @@ namespace shared_model {
                     validation::FieldValidator>>,
             validation::BatchOrderValidator> &validator);
 
-    types::SharedTxsCollectionType TransactionSequence::transactions() const {
-      return transactions_;
-    }
-
-    TransactionSequence::TransactionSequence(
-        const types::SharedTxsCollectionType &transactions)
-        : transactions_(transactions) {}
-
-    types::BatchesType TransactionSequence::batches() const {
-      if (batches_) {
-        return batches_.value();
-      }
-
-      std::unordered_map<std::string, std::vector<std::shared_ptr<Transaction>>>
-          extracted_batches;
-      std::vector<types::SharedTxsCollectionType> batches;
-      for (const auto &tx : transactions_) {
-        if (auto meta = tx->batchMeta()) {
-          auto hashes = meta.get()->transactionHashes();
-          auto batch_hash = this->calculateBatchHash(hashes);
-          extracted_batches[batch_hash].push_back(tx);
-        } else {
-          batches.emplace_back(std::vector<std::shared_ptr<Transaction>>{tx});
+    const types::SharedTxsCollectionType &TransactionSequence::transactions()
+        const {
+      if (not transactions_) {
+        types::SharedTxsCollectionType result;
+        auto transactions_amount = 0u;
+        for (const auto &batch : batches_) {
+          transactions_amount += batch.transactions().size();
         }
+        result.reserve(transactions_amount);
+        for (const auto &batch : batches_) {
+          auto transactions = batch.transactions();
+          std::copy(transactions.begin(),
+                    transactions.end(),
+                    std::back_inserter(result));
```

Comment:
Is it possible to rework it with move semantic?

Reply:
We have ideas to make all data objects move-only.

Reply:
I would better make transactions to be reference. Then we can avoid redundant copy of the vector. Another 'chair to seat' is to use `boost::back_move_inserter(result)`, but first option is easier IMO